### PR TITLE
Code Coverage Improvements

### PR DESCRIPTION
**Pull request by tylerflex**
Goal: approach 100% test coverage:

Started at 75% or so coverage.
As of Sept 28 12pm CEST at 86%.
As of Sept 29 2pm CEST at 89%.

## Running Tests
To get coverage report:

```
pip install pytest-cov
```
if not already installed

To run coverage tests with results printed to STDOUT:
```
pytest tests --cov-report term-missing --cov=tidy3d
```
To run coverage tests and get output as .html (more intuitive):
```
pytest tests --cov-report=html --cov=tidy3d
open htmlcov/index.html
```

## Things that haven't been fully tested.
- [ ] Data (Tyler, will change soon)
- [ ] Base (Tyler, will change soon)
- [ ] Some `PolySlab` operations that I don't understand.(Weiliang)
- [ ] A good chunk of the webapi (Lei Z)
- [x] Mode solver plugin (Momchil)
- [ ] Dispersion fitter (Weiliang)
- [ ] Updater (Momchil)
- [ ] Polygon RTree Stuff (Momchil)

## Notes on what could be improved
- [ ] `.surfaces` logic regarding `_normal_index`.
- [ ] Some of the source logic regarding `_dir_vec`.
- [ ] Monitor time mesh generation.
- [ ] `ArrayLike` types, define some consisten behavior or just figure out how to validate numpy arrays properly,
- [ ] webapi is a huge mess.
- [ ] reorganize the tests mainly by source file.
- [ ] PolySlab operations are very difficult to follow, perhaps unavoidable.

### CODE REVIEW

**Comment by tylerflex** [comment]
@momchil-flex I'd be in favor (if it makes sense to) of merging this before 1.7.  I assigned everything missing as individual issues and we can get those in individually at later times. (Doesn't have to all be in one PR).

**Comment by tylerflex** [comment]
@momchil-flex I'd be in favor of merging this before 1.7 if it makes sense to. Basically don't see a huge benefit to waiting.  I assigned everything missing as individual issues and we can get those in individually at later times. (Doesn't have to all be in one PR). Note also that I organized all the test files into a few directory sections to make it more organized.

**Comment by tylerflex** [comment]
@momchil-flex I just moved most of the tests from `test_components.py` to their corresponding file (ie `test_simulation.py` etc). The few remaining tests were testing `Tidy3DBaseModel` so I renamed it to `test_base.py`. Squashed everything to one commit. Think we could merge this soon. The other coverage things (eg from weiliang) we can add as separate PRs later?

File: tidy3d/components/validators.py [hunk 1]
```diff
@@ -108,18 +108,13 @@ def check_symmetry(cls, val, values):
     return check_symmetry
 
 
-def assert_unique_names(field_name: str, check_mediums=False):
+def assert_unique_names(field_name: str):
     """makes sure all elements of a field have unique .name values"""
 
     @pydantic.validator(field_name, allow_reuse=True, always=True)
     def field_has_unique_names(cls, val, values):
         """check for intersection of each structure with simulation bounds."""
```
Comment (momchil-flex) [inline tidy3d/components/validators.py:116]: This comment seems wrong here

File: tidy3d/components/simulation.py [hunk 2]
```diff
@@ -600,14 +594,14 @@ def _validate_monitor_size(self) -> None:
...
-        if self.run_time <= 0:
-            raise SetupError(
-                "The `Simulation.run_time` parameter was left at its default value of 0.0. "
-                "For running a simulation on our servers it must be set to > 0.0."
-            )
+    #     if self.run_time <= 0:
+    #         raise SetupError(
+    #             "The `Simulation.run_time` parameter was left at its default value of 0.0. "
+    #             "For running a simulation on our servers it must be set to > 0.0."
+    #         )
```
Comment (momchil-flex) [inline tidy3d/components/simulation.py:604]: Should we just remove this?

**Pull request merged**